Firestore/Example/Tests/Integration/FSTTransactionTests.mm

```diff
@@ -543,21 +523,13 @@ - (void)testHandleReadingOneDocAndWritingAnother {
         return nil;
       }
       completion:^(id _Nullable result, NSError *_Nullable error) {
-        // We currently require every document read to also be written.
-        // TODO(b/34879758): Add this check back once we drop that.
-        // NSError *error = nil;
-        // FIRDocument *snapshot = [transaction getDocument:doc1 error:&error];
-        // XCTAssertNil(error);
-        // XCTAssertEquals(0, tries);
-        // XCTAssertEqualObjects(@(1234), snapshot[@"count"]);
-        // snapshot = [transaction getDocument:doc2 error:&error];
-        // XCTAssertNil(error);
-        // XCTAssertEqualObjects(@(16), snapshot[@"count"]);
-        XCTAssertNotNil(error);
-        XCTAssertEqual(error.code, FIRFirestoreErrorCodeInvalidArgument);
+        XCTAssertNil(error);
+        XCTAssertEqual((int)(*counter), 2);
```

Comment:
This cast should be unnecessary.

If you do find a need to cast, prefer C++ casts to C casts. In this case the C++ cast would be `static_cast<int>(*counter)`.